partner_communication_switzerland/models/contracts.py

```diff
@@ -266,8 +266,33 @@ def _send_reminders_for_birthday_in_1day_or_2months(self):
         tomorrow = today + relativedelta(days=1)
         sponsorships_with_birthday_tomorrow = \
             self._get_sponsorships_with_child_birthday_on(tomorrow)
+
+        sponsorships_to_send_reminder = []
+
+        for sponsorship in sponsorships_with_birthday_tomorrow:
+            sponsorship_correspondences = self.env['correspondence'].search([
+                ('sponsorship_id', '=', sponsorship.id),
+                ('direction', '=', "Supporter To Beneficiary"),
+                ('sent_date', '!=', False),
+                ('sent_date', '<=', str(datetime.now())),
+                ('sent_date', '>=', str(datetime.now() - relativedelta(months=-2)))
+            ])
+
+            if sponsorship_correspondences:
+                sponsorships_to_send_reminder.append(sponsorship_correspondences)
+
+            sponsorship_gifts = self.env['sponsorship.gift'].search([
+                ('sponsorship_id', '=', sponsorship.id),
+            ])
+
+            for gift in sponsorship_gifts:
+                for invoice_line in gift.invoice_line_ids:
+                    if str(datetime.now()) >= invoice_line.last_payment \
+                            >= str(datetime.now() + relativedelta(months=-2)):
+                        sponsorships_to_send_reminder.append(gift.sponsorship_id)
```

Comment:
Instead of looping through all the gifts you could directly add the filter in the search like you did with the letters. Use the field `date_partner_paid` in the gift object.

Reply:
`sponsorships_to_avoid += sponsorship`